google/cloud/spanner_v1/client.py

```diff
@@ -173,19 +174,27 @@ def __init__(
         client_options=None,
         query_options=None,
     ):
+        self._emulator_host = _get_spanner_emulator_host()
+
+        if client_options and type(client_options) == dict:
+            self._client_options = google.api_core.client_options.from_dict(
+                client_options
+            )
+        else:
+            self._client_options = client_options
+
+        if self._emulator_host:
+            credentials = AnonymousCredentials()
+        elif isinstance(credentials, AnonymousCredentials):
+            self._emulator_host = self._client_options.api_endpoint
```

Comment:
I am a bit unsure about the logic here: if SPANNER_EMULATOR_HOST is not set and an AnonymousCredentials is provided, then we set `_emulator_host` as `api_endpoint`. 

I think copying from `api_endpoint` to`_emulator_host` is redundant. If I specify `AnonymousCredentials` but not giving an `api_endpoint`, it will raise an error because the production backend does not support it. 

Why do we need this?

Reply:
So the point of this is so that the correct gRPC channel is created. When using the emulator, `insecure_channel` needs to be used. We use `secure_channel` by default. So we check `self._emulator_host` to know which channel type to use.

Reply:
Okay. I see your point. In Go, we directly give `api_endpoint`, `AnonymousCredentials` and `insecure channel` together: https://github.com/googleapis/google-cloud-go/blob/master/spanner/admin/database/apiv1/init.go#L28-L33.